src/babylon.scene.ts

```diff
@@ -830,29 +830,46 @@
          * @see BABYLON.Animatable
          * @see http://doc.babylonjs.com/page.php?p=22081
          */
-        public beginAnimation(target: any, from: number, to: number, loop?: boolean, speedRatio: number = 1.0, onAnimationEnd?: () => void, animatable?: Animatable): Animatable {
-
-            this.stopAnimation(target);
-
+        public beginAnimation(target: any, from: number, to: number, loop?: boolean, speedRatio: number = 1.0, onAnimationEnd?: () => void, animatable?: Animatable,transitionSpeed: number = 1.0 ): Animatable {
+    
+    
+     //////use Scene.GetCurrentAnimatableByTarget(target) to get current primary playing animatable for target;
+      
+     
+          //  this.stopAnimation(target);
+            var transitionFunction = null;
+            if(transitionSpeed < 1)  {
+                                       transitionFunction = new BABYLON.LinearFADE(transitionSpeed,this._engine);
+                                     }
+              
             if (!animatable) {
-                animatable = new Animatable(this, target, from, to, loop, speedRatio, onAnimationEnd);
+                animatable = new Animatable(this, target, from, to, loop, speedRatio, onAnimationEnd,animatable,transitionFunction);
+                //animatable.transitionFunction = transitionFunction;
             }
-
             // Local animations
             if (target.animations) {
                 animatable.appendAnimations(target, target.animations);
-            }
-
+               
+           }
             // Children animations
             if (target.getAnimatables) {
                 var animatables = target.getAnimatables();
                 for (var index = 0; index < animatables.length; index++) {
-                    this.beginAnimation(animatables[index], from, to, loop, speedRatio, onAnimationEnd, animatable);
+                    this.beginAnimation(animatables[index], from, to, loop, speedRatio, onAnimationEnd, animatable,transitionSpeed);
                 }
             }
 
             animatable.reset();
 
+              var maxAnimations = 2;

```

Comment:
for now we want to have max
t 2 animatable per target,,,, i can remove this variable if u want